x/valset/keeper/keeper.go

```diff
@@ -516,12 +527,19 @@ func (k Keeper) GetSigningKey(ctx sdk.Context, valAddr sdk.ValAddress, chainType
 }
 
 // IsJailed returns if the current validator is jailed or not.
-func (k Keeper) IsJailed(ctx sdk.Context, val sdk.ValAddress) bool {
-	return k.staking.Validator(ctx, val).IsJailed()
+func (k Keeper) IsJailed(ctx context.Context, val sdk.ValAddress) bool {
+	a, err := k.staking.Validator(ctx, val)
+	if err != nil {
+		panic(err)
+	}
+	return a.IsJailed()
 }
 
-func (k Keeper) Jail(ctx sdk.Context, valAddr sdk.ValAddress, reason string) error {
-	val := k.staking.Validator(ctx, valAddr)
+func (k Keeper) Jail(ctx context.Context, valAddr sdk.ValAddress, reason string) error {
+	val, err := k.staking.Validator(ctx, valAddr)
+	if err != nil {
+		return err
+	}
 	if val == nil {
 		return ErrValidatorWithAddrNotFound.Format(valAddr)
 	}
```

Comment:
This check is no longer needed. The `Validator()` handles validator not found error.

Reply:
Is it ok to remove the code then?

Reply:
yes